pkg/apis/nmstate/v1alpha1/condition_types_test.go

```diff
@@ -0,0 +1,98 @@
+package v1alpha1
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Conditions", func() {
+	Context("when the list is empty", func() {
+		It("should return nil when finding a condition", func() {
+			condition := ConditionList{}.Find(NodeNetworkStateConditionFailing)
+			Expect(condition).To(BeNil())
+		})
+	})
+
+	Context("when composed by one condition with known type and unknown status", func() {
+		originalConditions := ConditionList{
+			Condition{
+				Type:    NodeNetworkStateConditionFailing,
+				Status:  corev1.ConditionUnknown,
+				Reason:  "foo",
+				Message: "bar",
+				LastHeartbeatTime: metav1.Time{
+					Time: time.Unix(0, 0),
+				},
+				LastTransitionTime: metav1.Time{
+					Time: time.Unix(0, 0),
+				},
+			},
+		}
+
+		Context("and we add different condition", func() {
+			conditions := originalConditions.DeepCopy()
+			conditionType := NodeNetworkStateConditionAvailable
+
+			It("should add new condition", func() {
+				conditions.Set(conditionType, corev1.ConditionTrue, "foo", "bar")
+				Expect(len(conditions)).To(Equal(len(originalConditions) + 1))
+
+				condition := conditions.Find(conditionType)
+				Expect(condition.Type).To(Equal(conditionType))
+				Expect(condition.Status).To(Equal(corev1.ConditionTrue))
+				Expect(condition.Reason).To(Equal("foo"))
+				Expect(condition.Message).To(Equal("bar"))
+				Expect(condition.LastHeartbeatTime).To(Equal(condition.LastTransitionTime))
+			})
+		})
+
+		Context("and we update the condition with same values", func() {
+			conditionType := NodeNetworkStateConditionFailing
+
+			It("should change LastHearbeatTime", func() {
+				conditions := originalConditions.DeepCopy()
+				conditions.Set(conditionType, corev1.ConditionFalse, "foo", "bar")
+
+				By("Shouldn't add new condition")
+				Expect(len(conditions)).To(Equal(len(originalConditions)))
+
+				condition := conditions.Find(conditionType)
+				originalCondition := originalConditions.Find(conditionType)
+
+				By("Should change LastHeartbeatTime")
+				Expect(originalCondition.LastHeartbeatTime.Time.Before(condition.LastHeartbeatTime.Time)).To(BeTrue(), "LastHeartbeatTime of updated condition wasn't updated")
+			})
+		})
+
+		Context("and we update the condition with different values", func() {
```

Comment:
Usa Table Driven Tests here to test all the sensitive fields independently

Reply:
This PR is just moving the test module from one place to another. I'm not a big fan of test matrix unless it really makes sense to run the same test against a list of data. Feel free to suggest follow-up PR with example of text matrix. I feel like it is better to keep it slim and readable.